src/interfaces/IAccountLoupe.sol

```diff
@@ -3,41 +3,48 @@ pragma solidity ^0.8.25;
 
 import {ModuleEntity} from "../interfaces/IModuleManager.sol";
 
-/// @notice Pre and post hooks for a given selector.
-/// @dev It's possible for one of either `preExecHook` or `postExecHook` to be empty.
-struct ExecutionHook {
-    ModuleEntity hookFunction;
-    bool isPreHook;
-    bool isPostHook;
+// Represents data associated with a specifc function selector.
+struct ExecutionDataView {
+    // The module that implements this execution function.
+    // If this is a native function, the address must remain address(0).
+    address module;
+    // Whether or not the function needs runtime validation, or can be called by anyone. The function can still be
+    // state changing if this flag is set to true.
+    // Note that even if this is set to true, user op validation will still be required, otherwise anyone could
+    // drain the account of native tokens by wasting gas.
+    bool isPublic;
+    // Whether or not a global validation function may be used to validate this function.
+    bool allowGlobalValidation;
+    // The execution hooks for this function selector.
+    bytes32[] executionHooks;
```

Comment:
These should probably have the `HookConfig` type, so the caller can interpret them in the same way as they're used in the install interfaces

Reply:
Yeah, I was thinking and choose to do the conversion on the SDK to avoid loops on the contract. Maybe it is worth it for callers not using our SDK.

Reply:
Ah I see. I think we can merge the two loops if we want (loop 1 within `.values()`, and loop 2 in the loupe function). Left a comment with details.